### PR TITLE
2022 03 23 updating shiny dashboards from excel via mongodb the good the bad and the shiny

### PR DESCRIPTION
**Pull request by MarkMc1089**
It seems the last PR did not take all the changes. I've just pushed an update to the README that details how to avoid one issue when running the dashboard (there was a file expected that is not present in the repo, since it has a risk of pushing real data). Do you want to give it another go @MWilson92?

### CODE REVIEW

**Comment by MarkMc1089** [comment]
For the 2 conflicts, can take the new versions. I'm not sure how to make GitHub do this tho, do you?

**Comment by MWilson92** [comment]
> For the 2 conflicts, can take the new versions. I'm not sure how to make GitHub do this tho, do you?

no, i don't think it can be done in GitHub. I've had a look at doing in GitHub desktop and had no success there either. I think you'll have to manually identify the issues and resolve them. I think easiest way to deal with the XLSM might be just to remove it entirely and then do another PR to put back in. Is this data ok to be public? 

The merge conflit for the readme should show in RStudio and can resolve from there

**Comment by MarkMc1089** [comment]
I've resolved conflicts. But pls hold off on accepting the PR. I managed to get the binder built, but got the same issue with using highcharter as you. I've tried something out to see if it fixes things in this branch. I'll see if it does work...unfortunately I'm getting the nginx 503 error at present.

**Comment by MarkMc1089** [comment]
> Is this data ok to be public?

Yes, it is all made up data.

**Comment by MarkMc1089** [comment]
The binder seems to be built OK...

```
Successfully pushed 3i2li627.gra7.container-registry.ovh.net/binder/ovhbhub-r2d-g5b5b759nhsbsa-2ddata-2danalytics-2dcoffee-2dand-2dcoding-879e10:80f22a6ed45701bdeb762fb8551ae6571ccc9984
```

... but the loading spinner is still going. It seems to have been a long time just like this.

image

**Comment by MarkMc1089** [comment]
Well that did not fix it. Now getting a 404 error when the binder tries to load. I'll revert back to the original config files and try something else!

**Comment by MarkMc1089** [comment]
Problem resolved by adding `libglpk-dev` to be apt-get installed. No idea why it used to work without this. Possibly due to some package update. Ready for PR to be merged, thanks.

File: 2022-03-23 Updating Shiny dashboards from Excel via MongoDB (the Good, the Bad and the Shiny)/README.md [hunk 1]
```diff
@@ -6,7 +6,25 @@
...
+# Ensure you are in the project root folder
+sample_data <- read_excel("data-raw/nhs_pensions_FIC_data.example.xlsm")
+# You will need to source the data_prep.R file in the R folder
+data_prep(sample_data)
+```
+
+To run the app, just open `app.R` and click Run app in RStudio.
+=======
 an extension of [`{nhsbsaShinyR}`](https://github.com/nhsbsa-data-analytics/nhsbsaShinyR) template developed by NHS Business Services Authority Data Analytics Learning Lab to use as a template for building NHSBSA branded R `{shiny}` dashboards. 
+>>>>>>> 60adb8509e3444283a9124d8dbedbae4639eb756
```
Comment (MWilson92) [inline 2022-03-23 Updating Shiny dashboards from Excel via MongoDB (the Good, the Bad and the Shiny)/README.md:27]: merge conflict is still in README. need to remove `<<<<<<<<<< HEAD`, and everything between `========` and `>>>>>>` inclusive.

**Review by MWilson92 — CHANGES_REQUESTED** [review]

File: 2022-03-23 Updating Shiny dashboards from Excel via MongoDB (the Good, the Bad and the Shiny)/README.md (hunk 1 above)
Comment (MarkMc1089) [reply 2022-03-23 Updating Shiny dashboards from Excel via MongoDB (the Good, the Bad and the Shiny)/README.md]: Done. That's what I get for editing files directly in GitHub!

**Comment by MarkMc1089** [comment]
Just so you are aware, although the code runs fine now, there is an issue with connecting to mongoDB. I'll look into it and make a (final I hope) PR once working. This whole binder business has not been straightforward!

**Review by MWilson92 — APPROVED** [review]
LGTM!

**Pull request merged**